### ISSUE TITLE
Correct way to get a cognito identity inside of a lambda using Amplify, API Gateway IAM_AUTH

### ISSUE THREAD

**Issue by gregcubet3000** [aws-amplify/amplify-js#3658]
** Which Category is your question related to? **
Auth and lambdas

** What AWS Services are you utilizing? **
It is a basic website that uploads images to an s3 bucket.  End users should be able to upload and see their own images, editors should be able to see anyones images. 

1. It is a React website, 
2. using AWS Amplify's  API.post library to post using Sig v4  to 
3. API Gateway using IAM_AUTH authorization
4. to connect to an AWS lambda (integration type Lambda, no header mappings)
5. that processes and posts to an encrypted S3 bucket.

I am using a federated userpool with google, fb, and cognito.

** Provide additional details e.g. code snippets **

Design issue 1: Authorizing users separately based on who they are
--------------------------------------------------------------------
I have been going around in circles on this.  I have authentication sorted but I want to authorize certain parts of my lambda based on who you are.   E.g. certain members of my team should be able to see files that a customer submits to the website. 

Design issue 2: Determining identity of user in lambda itself
--------------------------------------------------------------------
I also need to organise the data I'm writing in my lambda per user basis, so I need to see the identity of the user inside of the lambda.   I can happily get this on the client side from the amplify libraries.

  Auth.currentAuthenticatedUser().then(user => {  /// then call API.Post with user.id } )

But I would like to somehow derive this information on the server side as well without relying on passing it through from the client side.  

I have messed around with API gateway a bit to try and pass more info through but have not been successful yet. 

This is the closest info I have found but I would love a better nudge in the right direction. 
https://stackoverflow.com/questions/31658816/aws-api-gateway-how-to-pass-iam-identity-to-lambda-function

**Comment by haverchuck** [comment]
@gregcubet3000  - In the Lambda, have you tried looking at `event. requestContext,identity`?  You can get the cognito identity pool 'identity' for the user in that object, I believe.

**Comment by gregcubet3000** [comment]
I think I need to map something in API gateway, because it is not coming through by default?   I was thinking that maybe my settings in API gateway need to be tweaked.

![image](https://user-images.githubusercontent.com/52939501/61329418-99130c80-a860-11e9-894b-95ed953940bc.png)

I console.log(util.inspect(event)) and I don't have it in my lambda...   I was thinking a combo of 'invoke with caller credentials' and a header mapping...  any help in the right direction would be great.

**Comment by haverchuck** [comment]
@gregcubet3000 Did you setup the API / Lambda using the Amplify CLI?

**Comment by gregcubet3000** [comment]
No... my project is so old that I can't seem to get the Amplify CLI up and running...  We started the project when Amplify was very unbaked and I'm having a hard time converting.   I will have to post a separate q about that...

I think I finally got it.  I added a mapping template that passes through the headers...   I added application/json to mapping templates and then used the built in 'method request passthrough' template.    

![image](https://user-images.githubusercontent.com/52939501/61332672-2c037500-a868-11e9-8fe5-68e4bdb42cc3.png)

Now in the lambda using console.log(util.inspect(event)) I am getting full headers, which is giving me event.context['cognito-identity-id']
